docs/DeveloperGuide.md

```diff
@@ -441,43 +441,107 @@ considering code quality and end user experience.
 
 Step 4. `UpdateCommand#execute` then checks if a `Product` of the same `Name` as the `Product` to be updated exists in the `Set<Product>`.
 If the `Product` exists, the method does an additional check to ensure that either the tag(s) or quantity (or both)
-is supplied for the `Product` to be updated, failing which, an exception is thrown. If the check passes, the original
+is supplied in the `UpdateProductDescriptor`, failing which, an exception is thrown. If the check passes, the original
 `Product` is removed from the set.
 
-Step 5. `UpdateCommand#execute` adds the updated `Product` to the `Set<Product>`, and creates an updated
-warehouse/supplier with the updated product. The method then calls `Model#setWarehouse` or `Model#setSupplier` to update the model,
-and calls `Model#updateFilteredWarehouseList` to update the list to be displayed to the user.
-The method then passes a `CommandResult` with a success message back to `LogicManager#execute`. Finally, the model
+Step 5. `UpdateCommand#createUpdatedProduct` then creates a new product based on the product name and `UpdateProductDescriptor`. The `execute` method then adds the updated `Product` to the `Set<Product>`, and creates an updated
+warehouse/supplier with the updated product. The method then updates the model with the edited warehouse/supplier, and the `FilteredWarehouseList` to be displayed to the user.
+The method then passes a `CommandResult` with a success message back to the `LogicManager`. Finally, the model
 is saved and the GUI is updated with the success message.
 
-The following activity diagram summarizes what happens when a user updates a product: (TODO: Insert Diagram)
+The following activity diagram summarizes what happens when a user updates a product:
+![Update Product Command Activity Diagram](images/UpdateCommandActivityDiagram.png)
+
+### Assign macro feature
+
+#### What the assign macro feature does
+
+The assign macro feature allows users to be able to create their own alias for a specific command string which can be used to enter commands after the macro is saved.
+
+#### How it is implemented
+
+The assign macro mechanism is facilitated by 2 components: `AssignMacroCommandParser` and `AssignMacroCommand`.
+`AssignMacroCommandParser`'s job is to parse the user input to generate the correct objects for the `AssignMacroCommand`.
+`AssignMacroCommand`'s job is to execute the main logic for updating the model with the new macro.
+
+Given below is an example usage scenario, together with a sequence diagram, to show how the assign macro mechanism behaves at each step.
+![Assign Macro Command Sequence Diagram](images/AssignMacroCommandSequenceDiagram.png)
+
+Step 1. The user discovers that he frequently updates a specific warehouse and decides to create a new macro with the alias "uwm" for the command string "update ct/w n/MainWarehouse" so that he can enter a shorter command in the future.
```

Comment:
Maybe he/she here would be better 🤔

Reply:
Or a suggested phrasing could be:

The user frequently updates a specific warehouse and decides to create a new macro with the alias "uwm" for the command string "update ct/w n/MainWarehouse" so as to shorten subsequent command inputs.